prog/types.go

```diff
@@ -81,6 +82,68 @@ func (f *Field) Dir(def Dir) Dir {
 	return def
 }
 
+type ArgFinder func(path []string) Arg
```

Comment:
Do ArgFinder, SquashedArgFound and Expression need to be exported from the package?